### PR TITLE
[PyOV] Add Range-4 as part of opset12

### PR DESCRIPTION
**Pull request by jiwaszki**
### Details:
 - Add `opset12.range` as it was missing since `opset4`.

### Tickets:
 - *116684*

### CODE REVIEW

File: src/bindings/python/tests/test_graph/test_sequence_processing.py [hunk 1]
```diff
@@ -5,27 +5,47 @@
...
+def test_range_1():
     start = 5
     stop = 35
     step = 5
 
-    node = ov.range(start, stop, step)
+    node = ov.opset11.range(start, stop, step)
     assert node.get_output_size() == 1
     assert node.get_type_name() == "Range"
     assert list(node.get_output_shape(0)) == [6]
```
Comment (mitruska) [inline src/bindings/python/tests/test_graph/test_sequence_processing.py:32]: Worth to add expected output type also for v0::range test.

File: src/bindings/python/src/openvino/runtime/opset12/__init__.py [hunk 2]
```diff
@@ -120,7 +120,7 @@
 from openvino.runtime.opset1.ops import prior_box_clustered
 from openvino.runtime.opset1.ops import psroi_pooling
 from openvino.runtime.opset4.ops import proposal
-from openvino.runtime.opset1.ops import range
+from openvino.runtime.opset12.ops import range
```
Comment (mitruska) [inline src/bindings/python/src/openvino/runtime/opset12/__init__.py:123]: If `def range` will be moved to `opset4/ops.py`:
```suggestion
from openvino.runtime.opset4.ops import range
```

**Review by mitruska — APPROVED** [review]
Approved current version, but left few suggestions to consider (can be discussed and applied separately).

File: src/bindings/python/tests/test_graph/test_sequence_processing.py (hunk 1 above)
Comment (jiwaszki) [reply src/bindings/python/tests/test_graph/test_sequence_processing.py:32]: Tbh I do not want to change this test now.

File: src/bindings/python/src/openvino/runtime/opset12/ops.py [hunk 3]
```diff
@@ -118,3 +119,29 @@ def group_normalization(
             "epsilon": epsilon,
         },
     )
+
+
+@nameable_op
+def range(
```
[the remark this answers is not in the document]
Comment (jiwaszki) [reply src/bindings/python/src/openvino/runtime/opset12/ops.py]: Done

File: src/bindings/python/src/openvino/runtime/opset12/__init__.py (hunk 2 above)
Comment (jiwaszki) [reply src/bindings/python/src/openvino/runtime/opset12/__init__.py]: Done

**Pull request merged**